### PR TITLE
Include Postgres schema dumping into make schema

### PR DESCRIPTION
**Pull request by duduribeiro**
**What this PR does / why we need it**:

After making some change in the database we need to regenerate the
schema for Postgres. This changes the `make schema` command to
accept a env var `POSTGRES_DATABASE_URL` and uses it to migrate
the database and updates the schema after the migration

**Verification steps** 
Start your postgres database and create the data.

to simulate a non-executed migration in postgres you can run:
`DATABASE_URL="YOUR_PG_URL" bundle exec rake db:rollback db:schema:dump`

the db/postgres_schema.rb should have been changed with some columns being removed.

Now you can run `POSTGRES_DATABASE_URL="YOUR_PG_URL" make schema` (it will run the migration and redump the schema) and your
db/postgres_schema.rb should have no difference (it will be the same as git since we regenerated the schema)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,6 +26,7 @@ C:\\nppdf32Log\\debuglog.txt
 .zeus.sock
 .deploy
 .s3cfg
+.tool-versions
```
Comment (duduribeiro) [inline .gitignore]: This is for https://asdf-vm.com

File: Makefile [hunk 2]
```diff
@@ -187,6 +180,29 @@ oracle-database:
...
+	MASTER_PASSWORD=p USER_PASSWORD=p ORACLE_SYSTEM_PASSWORD=threescalepass NLS_LANG='AMERICAN_AMERICA.UTF8' DATABASE_URL="oracle-enhanced://rails:railspass@127.0.0.1:1521/systempdb" bundle exec rake db:drop db:create db:setup
+
+## Postgres database containerization
+postgres-database: ## Starts Postgres database container
+postgres-database:
+	[ "$(shell docker inspect -f '{{.State.Running}}' postgres-database 2>/dev/null)" = "true" ] || docker start postgres-database || docker run \
+		-p 5432:5432 \
+		--name postgres-database \
+		-e POSTGRES_USER=postgres \
+		-e POSTGRES_DB=threescale \
```
Comment (Martouta) [inline Makefile]: I think we don't need the password 🤔

File: Makefile [hunk 3]
```diff
@@ -187,6 +180,29 @@ oracle-database:
...
+		circleci/postgres:10.5-alpine
+
+postgres-db-setup: ## Creates databases in Postgres
+postgres-db-setup: postgres-database
+	DATABASE_URL="postgresql://postgres:@127.0.0.1:5433/threescale" bundle exec rake db:drop db:create db:setup
```
Comment (Martouta) [inline Makefile]: Mine works with: `DATABASE_URL="postgresql://postgres:@localhost:5433/circleci" bundle exec rake db:drop db:create db:setup` but if you tested this version and it works, good too 👍

File: Makefile [hunk 4]
```diff
@@ -187,6 +180,29 @@ oracle-database:
...
+
+schema: ## Runs db schema migrations. Run this when you have changes to your database schema that you have added as new migrations.
```
Comment (Martouta) [inline Makefile]: Probably this should do it also for mysql, so we do it for all DBs at once

File: Makefile [hunk 5]
```diff
@@ -187,6 +180,29 @@ oracle-database:
 		-v $(PWD)/script/oracle:/opt/oracle/scripts/setup \
 		quay.io/3scale/oracle:12.2.0.1-ee
 
+oracle-db-setup: ## Creates databases in Oracle
+oracle-db-setup: oracle-database
+	MASTER_PASSWORD=p USER_PASSWORD=p ORACLE_SYSTEM_PASSWORD=threescalepass NLS_LANG='AMERICAN_AMERICA.UTF8' DATABASE_URL="oracle-enhanced://rails:railspass@127.0.0.1:1521/systempdb" bundle exec rake db:drop db:create db:setup
+
+## Postgres database containerization
+postgres-database: ## Starts Postgres database container
```
Comment (hallelujah) [inline Makefile]: I wonder, most of the OS have postgres shipped with a package manager.
So is this really needed?

File: Makefile [hunk 6]
```diff
@@ -187,6 +180,29 @@ oracle-database:
...
+		-e POSTGRES_DB=threescale \
+		circleci/postgres:10.5-alpine
+
+postgres-db-setup: ## Creates databases in Postgres
+postgres-db-setup: postgres-database
+	DATABASE_URL="postgresql://postgres:@127.0.0.1:5433/threescale" bundle exec rake db:drop db:create db:setup
+
+schema: ## Runs db schema migrations. Run this when you have changes to your database schema that you have added as new migrations.
+	bundle exec rake db:migrate db:schema:dump
+	MASTER_PASSWORD=p USER_PASSWORD=p ORACLE_SYSTEM_PASSWORD=threescalepass NLS_LANG='AMERICAN_AMERICA.UTF8' DATABASE_URL="oracle-enhanced://rails:railspass@127.0.0.1:1521/systempdb" bundle exec rake db:migrate db:schema:dump
```
Comment (Martouta) [inline Makefile]: Do we need the `db:schema:dump` after the `db:migrate` ? 🤔

Comment (Martouta) [inline Makefile]: Does this work? In my case it requires me to send all these env vars: `DATABASE_URL="oracle-enhanced://rails:railspass@127.0.0.1:1521/systempdb" ORACLE_SYSTEM_PASSWORD=threescalepass NLS_LANG=AMERICAN_AMERICA.UTF8`

File: Makefile (hunk 5 above)
Comment (duduribeiro) [reply Makefile]: Just added if the person didn't want to install it  😄. I can remove it also and accept the DATABASE_URL as argument, so the user can pass it using any installation method used. Wdyt @hallelujah and @Martouta ?

File: Makefile (hunk 6 above)
Comment (duduribeiro) [reply Makefile]: good catch!. not necessary, tested here. thanks ☕️

File: Makefile (hunk 5 above)
Comment (Martouta) [reply Makefile]: Good for me 👍 and we follow the same approach for all DBs 😄 but i am opened to opinions about this 😄

File: Makefile [hunk 7]
```diff
@@ -171,6 +171,7 @@ oracle-db-setup: oracle-database
 schema: ## Runs db schema migrations. Run this when you have changes to your database schema that you have added as new migrations.
 	bundle exec rake db:migrate db:schema:dump
 	MASTER_PASSWORD=p USER_PASSWORD=p ORACLE_SYSTEM_PASSWORD=threescalepass NLS_LANG='AMERICAN_AMERICA.UTF8' DATABASE_URL="oracle-enhanced://rails:railspass@127.0.0.1:1521/systempdb" bundle exec rake db:migrate db:schema:dump
+	DATABASE_URL=$(POSTGRES_DATABASE_URL) bundle exec rake db:migrate
```
Comment (hallelujah) [inline Makefile]: We could have default one wdyt?

Comment (Martouta) [reply Makefile]: I agree with @hallelujah , it should work without params 😄

Comment (duduribeiro) [reply Makefile]: Agreed!. Just thinking which one can be the default 🤔 . any suggestion?

Comment (Martouta) [reply Makefile]: `postgresql://postgres:@localhost:5433/circleci`

Comment (duduribeiro) [reply Makefile]: Just don't know about `circleci` as the dbname.. This relies that the user setup their database naming `circleci`. Maybe `threescale`?. WDYT?

any suggestion @hallelujah ?

Comment (hallelujah) [reply Makefile]: Well I'd say the same as mysql :) `3scale_system_development`

**Comment by duduribeiro** [comment]
@Martouta and @hallelujah, updated the code

thanks

**Comment by duduribeiro** [comment]
also need review here @3scale/system 😄 

thanks

**Pull request merged**